aiohttp/cookiejar.py

```diff
@@ -121,34 +122,25 @@ def load(self, file_path: PathLike) -> None:
 
     def clear(self, predicate: Optional[ClearCookiePredicate] = None) -> None:
         if predicate is None:
-            self._next_expiration = ceil(time.time())
+            self._expire_heap.clear()
             self._cookies.clear()
             self._host_only_cookies.clear()
             self._expirations.clear()
             return
 
-        to_del = []
         now = time.time()
-        for (domain, path), cookie in self._cookies.items():
-            for name, morsel in cookie.items():
-                key = (domain, path, name)
-                if (
-                    key in self._expirations and self._expirations[key] <= now
-                ) or predicate(morsel):
-                    to_del.append(key)
-
-        for domain, path, name in to_del:
-            self._host_only_cookies.discard((domain, name))
-            key = (domain, path, name)
-            if key in self._expirations:
-                del self._expirations[(domain, path, name)]
-            self._cookies[(domain, path)].pop(name, None)
-
-        self._next_expiration = (
-            min(*self._expirations.values(), self.SUB_MAX_TIME) + 1
-            if self._expirations
-            else self.MAX_TIME
-        )
+        to_del = [
+            key
+            for (domain, path), cookie in self._cookies.items()
+            for name, morsel in cookie.items()
+            if (
+                (key := (domain, path, name)) in self._expirations
+                and self._expirations[key] <= now
```

Comment:
Are we assuming that clear() is not called often? Otherwise, it'd probably make more sense to have _expirations sorted and then be able to break early here instead of needing to iterate every one.

Reply:
The original issue does that clear() is being used, but I don't see how these changes would make clear() much faster. The heapq isn't being utilised here.

Reply:
I think it only would help if predicate is False and that code has been removed

Reply:
Right, I was thinking the predicate was used when a cookie is expiring, but it's the other way around.